### DISCUSSION TITLE
[Closed] ObservableObject (and other ObjectModel things)

### DISCUSSION THREAD

**Discussion by GalaxiaGuy** [CommunityToolkit/Maui#234] [New Feature Discussions]
Is there a plan to add things from here:
https://github.com/xamarin/XamarinCommunityToolkit/tree/main/src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel

Or instead, possibly direct people to the dotnet Community Toolkit, which has similar things:
https://github.com/CommunityToolkit/dotnet/tree/main/CommunityToolkit.Mvvm/ComponentModel

I think the second idea makes more sense since duplicating similar functionality seems a shame. However, the implementation of some things differ in important ways. The once I'm most interested in is that the dotnet Community Toolkit `ObservableObject` does not use a weak event for `PropertyChanged`.

I have opened an issue in that repo about it (and the fact that they don't have a `WeakEventManager`.
https://github.com/CommunityToolkit/dotnet/issues/80

(I also seem to discussing this in lots of repos now, but I think getting the best solution with things in the right place probably needs coordination between repos anyway).

**Comment by pictos** [comment]
Hey @GalaxiaGuy we don't pretend to support mvvm features (ObjectModel, ObersavableRangeCollection, etc) those will be included on CommunityToolkit.MVVM or .NET MAUI (like the weak event manager).
So if you see that XCT has something and CommunityToolkit.MVVM doesn't just open an issue or discussion there asking for the feature and ping us from XCT as well (:

**Comment by TheCodeTraveler** [comment]
In short, no we will not be adding `ObservableObject` to `CommunityToolkit.Maui`. Here is a link to the closed Proposal for `ObservableObject`: https://github.com/CommunityToolkit/Maui/issues/98

> The MVVM Toolkit offers ObservableObject: https://docs.microsoft.com/en-us/windows/communitytoolkit/mvvm/observableobject

Our recommendation is to use the [`CommunityToolkit.MVVM` NuGet package](https://www.nuget.org/packages/CommunityToolkit.Mvvm/).

**Reply by TheCodeTraveler** [reply]
_Adding `[Closed]` to the title_

**Comment by TheCodeTraveler** [comment]
Just FYI regarding `WeakEventManager` - we promoted `Xamarin.CommunityToolkit.Helpers.DelegateWeakEventManager` into the official .NET MAUI repo: https://github.com/dotnet/maui/issues/2703#issuecomment-966312097

**Comment by CliffAgius** [comment]
Closed as answered.
